### ISSUE TITLE
tf.nn.conv2d error message is inconsistent with documentation

### ISSUE THREAD

**Issue by cheyennee** [tensorflow/tensorflow#57167]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

TF2.4

### Custom Code

Yes

### Current Behaviour?

```shell
'SAME' or 'VALID' can be set to parameter 'padding' of tf.nn.conv2d  in documentation. However, the error message of tf.nn.conv2d shows that 'EXPLICIT' is also an allowd value of 'padding'. Actually, when 'padding' takes 'EXPLICIT', code doesn't work. So the error message is inconsistent with the documentation.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
results={}
try:
  arg_0 = tf.random.uniform([1, 8, 8, 1], dtype=tf.float32)
  arg_1 = tf.random.uniform([1, 1, 1, 3], dtype=tf.float32)
  strides = 1
  padding = "valid"
  results["res"] = tf.nn.conv2d(arg_0,arg_1,strides=strides,padding=padding,)
except Exception as e:
  results["err"] = "Error:"+str(e)
print(results)
'''
{'err': 'Error:Value for attr \'padding\' of "valid" is not in the list of allowed values: "SAME", "VALID", "EXPLICIT"\n\t; NodeDef: {{node Conv2D}}; Op<name=Conv2D; signature=input:T, filter:T -> output:T; attr=T:type,allowed=[DT_HALF, DT_BFLOAT16, DT_FLOAT, DT_DOUBLE, DT_INT32]; attr=strides:list(int); attr=use_cudnn_on_gpu:bool,default=true; attr=padding:string,allowed=["SAME", "VALID", "EXPLICIT"]; attr=explicit_paddings:list(int),default=[]; attr=data_format:string,default="NHWC",allowed=["NHWC", "NCHW"]; attr=dilations:list(int),default=[1, 1, 1, 1]> [Op:Conv2D]'}
'''
```

### Relevant log output

_No response_</details>

**Comment by tilakrayal** [comment]
@cheyennee,

_**tf.nn.conv2d**_ allows a list of explicit padding amounts to be passed to the padding parameter. This list is in the same format as what is passed to **_`tf.pad`_**, except the padding must be a nested list, not a tensor. For example, in the 2D case, the list is in the format `[[0, 0], [pad_top, pad_bottom], [pad_left, pad_right], [0, 0]] `when `data_format` is its default value of _`'NHWC'`_. The two [0, 0] pairs indicate the batch and channel dimensions have no padding, which is required, as only spatial dimensions can have padding.

```
inp = tf.ones((1, 3, 3, 1))
filter = tf.ones((2, 2, 1, 1))
strides = [1, 1]
padding = [[0, 0], [1, 2], [0, 1], [0, 0]]
output = tf.nn.conv2d(inp, filter, strides, padding=padding)
tuple(output.shape)

# Equivalently, tf.pad can be used, since convolutions pad with zeros.
inp = tf.pad(inp, padding)
# 'VALID' means to use no padding in conv2d (we already padded inp)
output2 = tf.nn.conv2d(inp, filter, strides, padding='VALID')
tf.debugging.assert_equal(output, output2)
```
Also please take a look at this doc [link](https://www.tensorflow.org/api_docs/python/tf/nn#explicit_padding_2) which delivers the required information. Thank you!

**Comment by cheyennee** [comment]
@tilakrayal  Thanks for reminding me. But the padding of tf.nn.conv2d can be set to 'SAME' or 'VALID' written on documentation https://tensorflow.google.cn/versions/r2.4/api_docs/python/tf/nn/conv2d#args. However, the error message shows that the padding can be set to 'SAME' or 'VALID' or 'EXPLICIT', which is inconsistent with the documentation.

**Comment by tilakrayal** [comment]
@cheyennee,
Here in the error trace EXPLICIT was considered as an attribute which is optional. Could you please take a look at this doc [link1](https://www.tensorflow.org/api_docs/cc/struct/tensorflow/ops/conv2-d/attrs#explicit_paddings_) and [link2](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/conv2-d) which provides the information. It might be difficult for the community if padding is EXPLICIT is not documented on the error message.

![image](https://user-images.githubusercontent.com/81610181/185608203-a551120d-300d-406a-bdd9-68881eaffba6.png)

Thank you!

**Comment by cheyennee** [comment]
@tilakrayal 
Get it! I may have misunderstood 'EXPLICIT' before. Now I really know the use of 'EXPLICIT'. Thank you very very much👍
